org.adempiere.base/src/org/compiere/model/MArchive.java

```diff
@@ -402,12 +410,26 @@ public File saveAsZip() {
 	 * @param Record_ID
 	 * @param trxName
 	 * @return int[], [0] = report count and [1] = document count
+	 * @deprecated - use {@link #getReportAndDocumentCountByRecordId(int, int, String, String)} instead
 	 */
 	public static int[] getReportAndDocumentCountByRecordId(int AD_Table_ID, int Record_ID, String trxName) {
+		return getReportAndDocumentCountByRecordId(AD_Table_ID, Record_ID, null, trxName);
+	}
+
+	/**
+	 * Get number of document and report archive by table and record id
+	 * 
+	 * @param AD_Table_ID
+	 * @param Record_ID

```

Comment:
this parameter is basically ignore and not doing anything here ...

Reply:
> this parameter is basically ignore and not doing anything here ...

It is still used for the case of Business Partner window.  When checking the archives in BP window, the query counts all the documents related to the current BP, independent of the table.

Reply:
ic, miss that. however, this become a typical case of you wouldn't know what the method parameter is for until you read through the code in it.